application-xpoll-api/src/test/java/com/xwiki/xpoll/internal/DefaultXPollManagerTest.java

```diff
@@ -0,0 +1,121 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.xpoll.internal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.inject.Provider;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.test.junit5.mockito.ComponentTest;
+import org.xwiki.test.junit5.mockito.InjectMockComponents;
+import org.xwiki.test.junit5.mockito.MockComponent;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.web.XWikiRequest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit tests for {@link DefaultXPollManager}.
+ *
+ * @version $Id$
+ * @since 2.1
+ */
+@ComponentTest
+public class DefaultXPollManagerTest
+{
+    private static final String XPOLL_SPACE_NAME = "XPoll";
+
+    private static final LocalDocumentReference XPOLL_CLASS_REFERENCE =
+        new LocalDocumentReference(XPOLL_SPACE_NAME, "XPollClass");
+
+    private static final LocalDocumentReference XPOLL_VOTES_CLASS_REFERENCE =
+        new LocalDocumentReference(XPOLL_SPACE_NAME, "XPollVoteClass");
+
+    @InjectMockComponents
+    private DefaultXPollManager manager;
+
+    @MockComponent
+    private Provider<XWikiContext> contextProvider;
+
+    private XWikiContext xWikiContext;
+
+    private XWikiRequest request;
+
+    private XWiki wiki;
+
+    private XWikiDocument document;
+
+    private BaseObject xpollObj;
+
+    @BeforeEach
+    void setup()
+    {
+        this.xWikiContext = mock(XWikiContext.class);
+        when(this.contextProvider.get()).thenReturn(this.xWikiContext);
+    }
+
+    @Test
+    void urlMultipleSpaces()
+    {
+        this.request = mock(XWikiRequest.class);
+        when(this.xWikiContext.getRequest()).thenReturn(this.request);
+        when(this.request.getContextPath()).thenReturn("xwiki");
+
+        DocumentReference docRef = new DocumentReference("XWiki", Arrays.asList("Space1", "Space2"), "Page");
+
+        String result = manager.getRestURL(docRef);
+        String expected = "xwiki/rest/wikis/XWiki/spaces/Space1/spaces/Space2/pages/Page/xpoll";
+        assertEquals(expected, result);
+    }
+
+    @Test
+    void getVoteResults() throws XWikiException
+    {
+        this.xpollObj = mock(BaseObject.class);
+        this.document = mock(XWikiDocument.class);
+        this.wiki = mock(XWiki.class);
```

Comment:
Either you make them local or, if you keep them as class fields, then annotate them with ``@Mock`` (easier).